homeassistant/components/powerfox/sensor.py

```diff
@@ -0,0 +1,150 @@
+
+    value_fn: Callable[[T], StateType]
+
+
+SENSORS_POWER: tuple[PowerfoxSensorEntityDescription, ...] = (
+    PowerfoxSensorEntityDescription[PowerMeter](
+        key="power",
+        translation_key="power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda meter: meter.power,
+    ),
+    PowerfoxSensorEntityDescription[PowerMeter](
+        key="energy_usage",
+        translation_key="energy_usage",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda meter: meter.energy_usage,
+    ),
+    PowerfoxSensorEntityDescription[PowerMeter](
+        key="energy_usage_low_tariff",
+        translation_key="energy_usage_low_tariff",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda meter: meter.energy_usage_low_tariff,
+    ),
+    PowerfoxSensorEntityDescription[PowerMeter](
+        key="energy_usage_high_tariff",
+        translation_key="energy_usage_high_tariff",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda meter: meter.energy_usage_high_tariff,
+    ),
+    PowerfoxSensorEntityDescription[PowerMeter](
+        key="energy_return",
+        translation_key="energy_return",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda meter: meter.energy_return,
+    ),
+)
+
+
+SENSORS_WATER: tuple[PowerfoxSensorEntityDescription, ...] = (
+    PowerfoxSensorEntityDescription[WaterMeter](
+        key="cold_water",
+        translation_key="cold_water",
+        native_unit_of_measurement=UnitOfVolume.CUBIC_METERS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda meter: meter.cold_water,
+    ),
+    PowerfoxSensorEntityDescription[WaterMeter](
+        key="warm_water",
+        translation_key="warm_water",
+        native_unit_of_measurement=UnitOfVolume.CUBIC_METERS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda meter: meter.warm_water,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PowerfoxConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Powerfox sensors based on a config entry."""
+    entities: list[SensorEntity] = []
+    for coordinator in entry.runtime_data:
+        if isinstance(coordinator.data, PowerMeter):
```

Comment:
if we have only one tuple with all entity descriptions (both water and power), could we simplify and remove this if/else here?

Reply:
You are right. I was misinterpreting the way the descriptions are being searched.